api/host_query_xjoin.py

```diff
@@ -279,8 +301,12 @@ def _query_filters(fqdn, display_name, hostname_or_id, insights_id, tags, stalen
             logger.debug("Adding id (uuid) to the filter list")
             hostname_or_id_filters += ({"id": {"eq": str(id)}},)
         query_filters = ({"OR": hostname_or_id_filters},)
+    elif provider_id:
+        query_filters = ({"provider_id": {"eq": provider_id}},)
     elif insights_id:
         query_filters = ({"insights_id": {"eq": insights_id}},)
+    elif provider_type:
+        query_filters = ({"provider_id": {"eq": provider_type}},)
```

Comment:
Noticed this after my review, but I'm pretty sure the filter name should be `provider_type`
```suggestion
        query_filters = ({"provider_type": {"eq": provider_type}},)
```

Reply:
fixed.